assets/js/modules/analytics-4/datastore/conversion-reporting.js

```diff
@@ -96,13 +124,116 @@ const actions = {
 	dismissLostConversionReportingEvents() {
 		return dismissLostConversionReportingEventsStore.actions.fetchDismissLostConversionReportingEvents();
 	},
+
+	/**
+	 * Stores conversion reporting inline data in the datastore.
+	 *
+	 * @since n.e.x.t
+	 * @private
+	 *
+	 * @param {Object} data Inline data, usually supplied via a global variable from PHP.
+	 * @return {Object} Redux-style action.
+	 */
+	receiveConversionReportingInlineData( data ) {
+		invariant( data, 'data is required.' );
+
+		return {
+			payload: { data },
+			type: RECEIVE_CONVERSION_REPORTING_INLINE_DATA,
+		};
+	},
+};
+
+export const reducer = createReducer( ( state, { payload, type } ) => {
+	switch ( type ) {
+		case RECEIVE_CONVERSION_REPORTING_INLINE_DATA: {
+			const { newEvents, lostEvents } = payload.data;
+
+			state.detectedEventsChange = { newEvents, lostEvents };
+			break;
+		}
+
+		default: {
+			break;
+		}
+	}
+} );
+
+export const selectors = {
+	/**
+	 * Checks whether the provided conversion reporting events are available.
+	 *
+	 * @since 1.135.0
+	 *
+	 * @param {Object}               state  Data store's state.
+	 * @param {string|Array<string>} events Conversion reporting events to check.
+	 * @return {(boolean|undefined)} True if all provided custom dimensions are available, otherwise false. Undefined if available custom dimensions are not loaded yet.
+	 */
+	hasConversionReportingEvents: createRegistrySelector(
+		( select ) => ( state, events ) => {
+			// Ensure events is always an array, even if a string is passed.
+			const eventsToCheck = Array.isArray( events ) ? events : [ events ];
+
+			const detectedEvents =
+				select( MODULES_ANALYTICS_4 ).getDetectedEvents();
+
+			if ( ! detectedEvents?.length ) {
+				return false;
+			}
+
+			return eventsToCheck.some( ( event ) =>
+				detectedEvents.includes( event )
+			);
+		}
+	),
+
+	/**
+	 * Gets all conversion reporting inline data from this data store.
+	 *
+	 * Not intended to be used publicly; this is largely here so other selectors can
+	 * request data using the selector/resolver pattern.
+	 *
+	 * @since n.e.x.t
+	 * @private
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {(Object|undefined)} Conversion reporting inline data.
+	 */
+	getConversionReportingEventsChange( state ) {
+		return state.detectedEventsChange;
+	},
+
+	/**
+	 * Checks if newEvents are present.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {boolean} TRUE if `newEvents` are present, FALSE otherwise.
+	 */
+	hasNewConversionReportingEvents:
+		hasConversionReportingEventsOfType( 'newEvents' ),
+
+	/**
+	 * Checks if lostEvents are present.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {boolean} TRUE if `lostEvents` are present, FALSE otherwise.
```

Comment:
Please also update the return type to show the potential to return `undefined`, similar to `getConversionReportingEventsChange`.